### ISSUE TITLE
Using a REStful update using PUT or PATCH instead of an operation

### ISSUE THREAD

**Issue by Healthedata1** [argonautproject/scheduling#27]
A REStful update using PUT or PATCH instead of an operation with the selected Appointment.  Technically the FHIR Server is doing more than storing but this could be used as well.

Pro: lighter weight.
Con: the Appointment is modified by the client ?

Others

**Comment by daliboz** [comment]
Also related to https://github.com/argonautproject/scheduling/issues/32 

If we switched this to PUT or PATCH, I assume GET would no longer be an option :)

**Comment by Healthedata1** [comment]
The current design assumes the Appointment is updated *only* by the
Server/Scheduler rather than both parties exchanging updated resources so
there is only a single *source of truth*.  This was how Cooper had laid it
out.   I thought about Patch too  - the server has to go through the same
business rules/processing before updating the Appointment.  Happy to
discuss - it doesn't change the status and state diagrams..

Eric

Eric M Haas, DVM, MS
Health eData Inc
211 South Jefferson Street, Napa, CA, 94559
707.227.2608|Skype: haas.eric1
ehaas@healthedatainc.com <ehaas@healhtedatainc.com>

**Comment by cooperthompson** [comment]
Is this issue (#27) about the Appointment availability query or the Appointment booking?  For Appointment availability, long term I think we need a POST operation, but short term I'm ok with just a GET.  For Appointment booking, I would think we'd want a POST not GET, since that isn't idempotent.

**Comment by cooperthompson** [comment]
We had a lot of discussion on the call today.  We clarified that the point of discussion wasn't GET vs. PUT/POST, but rather whether Appointment booking should be an Apportionment.interaction vs. a Appointment.$book operation.

Edit:  There was actually a question about GET, but we determined that since booking is not an idempotent operation (can lead to overbooking potentially), we'll stick with a POST.
